### PR TITLE
Hummingbird build of DevTools with a tabbar immitating the current tab layout.

### PR DESCRIPTION
**Pull request by DaveShuckerow**
I noticed an interesting behavior in the tests: The app looks right when switching between narrow and full-width mode when the window is ~800dp and it produces no errors at any size.

However in tests, the full-width mode will produce an overflow at for anything less than 1202dp width.

### CODE REVIEW

File: packages/devtools_flutter/lib/src/screen.dart [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  @override
+  void dispose() {
+    controller.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: buildAppBar(),
```
Comment (kenzieschmoll) [inline packages/devtools_flutter/lib/src/screen.dart]: won't the appbar end up being created for each screen? do we want that?

File: packages/devtools_flutter/pubspec.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
+name: devtools_flutter
+description: UI package for web-based performance tooling for Dart and Flutter written in Flutter.
+
+# Note: when updating this version, please update the corresponding entry in
+# lib/devtools.dart and ../devtools/pubspec.yaml as well to the same version.
+# When publishing new versions of this package be sure to publish a new version
+# of package:devtools as well. package:devtools contains a compiled snapshot of
+# this package.
```
Comment (kenzieschmoll) [inline packages/devtools_flutter/pubspec.yaml]: does this comment hold true?

File: packages/devtools_flutter/test/screen_test.dart [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import 'wrappers.dart';
+
+void main() {
+  const contentKey = Key('Page Content');
+  const content = SizedBox(key: contentKey);
+
+  group('Page widget', () {
```
Comment (kenzieschmoll) [inline packages/devtools_flutter/test/screen_test.dart]: nit: r/Page/Screen

File: packages/devtools_flutter/test/screen_test.dart [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    testWidgets('displays in narrow mode without error',
+        (WidgetTester tester) async {
+      await setWindowSize(const Size(800.0, 1200.0));
+
+      await tester.pumpWidget(wrap(const Screen(child: content)));
+      expect(find.byKey(contentKey), findsOneWidget);
+      expect(find.byKey(Screen.narrowWidth), findsOneWidget);
+      expect(find.byKey(Screen.fullWidth), findsNothing);
+    });
```
Comment (kenzieschmoll) [inline packages/devtools_flutter/test/screen_test.dart]: nit: trailing comma after `}` here and elsewhere

File: packages/devtools_flutter/test/wrappers.dart [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+/// Wraps [widget] with the build context it needs to load in a test.
+Widget wrap(Widget widget, {Size windowSize}) => MaterialApp(home: widget);
```
Comment (kenzieschmoll) [inline packages/devtools_flutter/test/wrappers.dart]: nit: remove `windowSize` param if it is not used, or instead use it in the return somehow

File: packages/devtools_flutter/lib/src/connect_screen.dart [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+          padding: EdgeInsets.only(left: 20.0),
+        ),
+        RaisedButton(
+          child: const Text('Connect'),
+          onPressed: connect,
+        ),
+      ],
+    );
+  }
+
```
[the remark this answers is not in the document]
Comment (DaveShuckerow) [reply packages/devtools_flutter/lib/src/connect_screen.dart]: Filling an issue to go with the TODO: https://github.com/flutter/devtools/issues/1112.

File: packages/devtools_flutter/lib/src/screen.dart [hunk 7]
```diff
@@ -0,0 +1,117 @@
+import 'package:flutter/material.dart';
+
+/// A page in the DevTools App, including the scaffolding and navigation tabs
```
[the remark this answers is not in the document]
Comment (DaveShuckerow) [reply packages/devtools_flutter/lib/src/screen.dart]: Done.

File: packages/devtools_flutter/lib/src/screen.dart [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+///   }
+/// }
+/// ```
+///
+/// For a sample implementation, see [ConnectPage].
+class Screen extends StatefulWidget {
+  static const Key narrowWidth = Key('Narrow Page');
+  static const Key fullWidth = Key('Full-width Page');
+
+  /// The width where we need to treat the page as narrow-width.
```
[the remark this answers is not in the document]
Comment (DaveShuckerow) [reply packages/devtools_flutter/lib/src/screen.dart]: Done.

File: packages/devtools_flutter/pubspec.yaml (hunk 2 above)
Comment (DaveShuckerow) [reply packages/devtools_flutter/pubspec.yaml]: No.  In fact, this package may end up being short-lived.

File: packages/devtools_flutter/test/screen_test.dart (hunk 3 above)
Comment (DaveShuckerow) [reply packages/devtools_flutter/test/screen_test.dart]: Done.

File: packages/devtools_flutter/test/screen_test.dart (hunk 4 above)
Comment (DaveShuckerow) [reply packages/devtools_flutter/test/screen_test.dart]: In test methods, the trailing comma doesn't improve the readability, and it only indents the body of the test method by an extra two spaces.

Also, no existing test code does this.

File: packages/devtools_flutter/test/wrappers.dart (hunk 5 above)
Comment (DaveShuckerow) [reply packages/devtools_flutter/test/wrappers.dart]: That's a good catch.  I ended up needing a separate windowSize setter that I added below.  Done.

**Comment by jacob314** [comment]
lgtm

**Comment by DaveShuckerow** [comment]
> If activating it through `flutter pub global`, we won't need the `pub global activate` line above.

Good suggestion, thanks.  Done.

**Pull request merged**